willCreateBlock-support-fullBlock

MarcusDenker opened the pull request:
willCreateBlock does not detect full blocks, this leads to to problems in the debugger that temps are read from the wrong context

MarcusDenker committed:
willCreateBlock does not detect full blocks, this leads to to problems in the debugger that temps are read from the wrong context

dionisiydk commented on src/Kernel-BytecodeEncoders/EncoderForSistaV1.class.st:
```diff
@@ -485,6 +485,15 @@ EncoderForSistaV1 class >> isCreateFullBlock: compiledBlock code: code at: pc [
+		  extensionsAt: pc
+		  in: code
+		  into: [ :extA :extB :nExtBytes | 
+		  (code at: pc + nExtBytes) = 16rF9 ]
```
Why we can't put this logic into isCreateBlockAt:in: ? It would be good to add a comment about it

MarcusDenker replied:
I am not yet at all sure where to put what.. I think it is good to have, on the level of the encoder, two different methods to check (ad we have two different ways to create blocks).

MarcusDenker commented:
StepThroughTest>>testStepThrough fails after this.

Which is interesting... and needs some analysis.

MarcusDenker committed:
Fix for the test, it was not doing a needed step...

MarcusDenker commented:
unix-64 / Tests-unix-64 / testStepOverAssignment – Unix64.Debugger.Tests.Base.AssignmentAndLiteralDebuggerTest

unix-64 / Tests-unix-64 / testNewDebugSession – Unix64.Debugger.Tests.Base.AssignmentAndLiteralDebuggerTest

unix-64 / Tests-unix-64 / testStepIntoAssignment – Unix64.Debugger.Tests.Base.AssignmentAndLiteralDebuggerTest

MarcusDenker committed:
No sure about these...

- step only once in testStepIntoAssignment and testStepOverAssignment. Why did we step twice?
- context of the block has size 2

Ducasse commented:
Marcus I integrate it and we will see.
